File: HolaMundoOfficeVB/Form1.vb
Imports Word = Microsoft.Office.Interop.Word
Imports Excel = Microsoft.Office.Interop.Excel

Public Class Form1
    Dim sfd As SaveFileDialog = New SaveFileDialog()

    Private Sub btnWord_Click(sender As Object, e As EventArgs) Handles btnWord.Click
        If sfd.ShowDialog() = DialogResult.OK Then
            Dim word As Word.Application = New Word.Application()
            Dim doc As Word.Document = word.Documents.Add()
            doc.Activate()
            word.Selection.TypeText(tbTexto.Text)
            doc.SaveAs2(sfd.FileName)
            word.Visible = True
        End If
    End Sub

    Private Sub btnExcel_Click(sender As Object, e As EventArgs) Handles btnExcel.Click
        If sfd.ShowDialog() = DialogResult.OK Then
            Dim excel As Excel.Application = New Excel.Application()
            Dim file As Excel.Workbook = excel.Workbooks.Add()
            Dim hoja As Excel.Worksheet = file.Worksheets.Add()
            file.Activate()
            hoja.Cells(1, 1).Value = tbTexto.Text
            file.SaveAs(sfd.FileName)
            excel.Visible = True
        End If
    End Sub
End Class

File: HolaMundoOffice/Form1.cs
using System;
using System.Windows.Forms;
using Excel = Microsoft.Office.Interop.Excel;
using Word = Microsoft.Office.Interop.Word;

namespace HolaMundoOffice
{
    public partial class Form1 : Form
    {
        SaveFileDialog sfd;

        public Form1()
        {
            sfd = new SaveFileDialog();
            InitializeComponent();
        }

        private void btnWord_Click(object sender, EventArgs e)
        {
            if(sfd.ShowDialog() == DialogResult.OK)
            {
                Word.Application word = new Word.Application();
                Word.Document doc = word.Documents.Add();
                doc.Activate();
                word.Selection.TypeText(tbTexto.Text);
                doc.SaveAs2(sfd.FileName);
                word.Visible = true;
            }
        }

        private void btnExcel_Click(object sender, EventArgs e)
        {
            if(sfd.ShowDialog() == DialogResult.OK)
            {
                Excel.Application excel = new Excel.Application();
                Excel.Workbook file = excel.Workbooks.Add();
                Excel.Worksheet hoja = file.Worksheets.Add();

                hoja.Cells.Range["A1"].Value = tbTexto.Text;
                file.SaveAs(sfd.FileName);
                excel.Visible = true;
            }
        }
    }
}
